lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -64,6 +64,48 @@ def initialize(target, installation_options)
         end
 
         class << self
+          # @param  [Xcodeproj::Project::Object::AbstractObject] object
+          #
+          # @return [Boolean] Whether input & output paths for the given object
+          #         should be stored in a file list file.
+          #
+          def input_output_paths_use_filelist?(object)
+            object.project.object_version.to_i >= 50
+          end
+
+          # Sets the input & output paths for the given script build phase.
+          #
+          # @param  [Xcodeproj::Project::Object::PBXShellScriptBuildPhase] phase
+          #         The phase to set input & output paths on.
+          #
+          # @param  [Hash] input_paths_by_config
+          #
+          # @return [Void]
+          def set_input_output_paths(phase, input_paths_by_config, output_paths_by_config)
+            if input_output_paths_use_filelist?(phase)
+              [input_paths_by_config, output_paths_by_config].each do |hash|
+                hash.each do |(real_path, _relative_path), files|
+                  generator = Generator::FileList.new(files)
+                  Xcode::PodsProjectGenerator::TargetInstallerHelper.update_changed_file(generator, real_path)
+                end
+              end
+
+              phase.input_paths = nil
+              phase.output_paths = nil
+              phase.input_file_list_paths = input_paths_by_config.each_key.map(&:last).uniq
+              phase.output_file_list_paths = output_paths_by_config.each_key.map(&:last).uniq
```

Comment:
another reason for some sort of `Struct` to model this - it's not very clear why `:last` is required, nor what it actually contains